code/datums/gamemodes/crash.dm

```diff
@@ -0,0 +1,289 @@
+			human_track = pick('sound/theme/neutral_hopeful1.ogg','sound/theme/neutral_hopeful2.ogg')
+		if(MODE_CRASH_DRAW_DEATH)
+			xeno_track = pick('sound/theme/nuclear_detonation1.ogg','sound/theme/nuclear_detonation2.ogg') 
+			human_track = pick('sound/theme/nuclear_detonation1.ogg','sound/theme/nuclear_detonation2.ogg')
+
+	for(var/i in GLOB.xeno_mob_list)
+		var/mob/M = i
+		SEND_SOUND(M, xeno_track)
+
+	for(var/i in GLOB.human_mob_list)
+		var/mob/M = i
+		SEND_SOUND(M, human_track)
+
+	for(var/i in GLOB.observer_list)
+		var/mob/M = i
+		if(ishuman(M.mind.current))
+			SEND_SOUND(M, human_track)
+			continue
+
+		if(isxeno(M.mind.current))
+			SEND_SOUND(M, xeno_track)
+			continue
+
+		SEND_SOUND(M, pick('sound/misc/gone_to_plaid.ogg', 'sound/misc/good_is_dumb.ogg', 'sound/misc/hardon.ogg', 'sound/misc/surrounded_by_assholes.ogg', 'sound/misc/outstanding_marines.ogg', 'sound/misc/asses_kicked.ogg'))
+
+	
+	log_game("[round_finished]\nGame mode: [name]\nRound time: [duration2text()]\nEnd round player population: [length(GLOB.clients)]\nTotal xenos spawned: [GLOB.round_statistics.total_xenos_created]\nTotal humans spawned: [GLOB.round_statistics.total_humans_created]")
+
+	announce_medal_awards()
+	announce_round_stats()
+	end_of_round_deathmatch()
+
+
+/datum/game_mode/crash/proc/initialize_xeno_leader()
+	var/list/possible_queens = get_players_for_role(BE_QUEEN)
+	if(!length(possible_queens))
+		return FALSE
+
+	var/found = FALSE
+	for(var/i in possible_queens)
+		var/datum/mind/new_queen = i
+		if(new_queen.assigned_role || is_banned_from(new_queen.current?.ckey, ROLE_XENO_QUEEN))
+			continue
+		if(queen_age_check(new_queen.current?.client))
+			continue
+		new_queen.assigned_role = ROLE_XENO_QUEEN
+		xenomorphs += new_queen
+		found = TRUE
+		break
+
+	return found
+
+/datum/game_mode/crash/proc/initialize_xenomorphs()
+	var/list/possible_xenomorphs = get_players_for_role(BE_ALIEN)
+	if(length(possible_xenomorphs) < xeno_required_num)
+		return FALSE
+
+	for(var/i in possible_xenomorphs)
+		var/datum/mind/new_xeno = i
+		if(new_xeno.assigned_role || is_banned_from(new_xeno.current?.ckey, ROLE_XENOMORPH))
+			continue
+		new_xeno.assigned_role = ROLE_XENOMORPH
+		xenomorphs += new_xeno
+		possible_xenomorphs -= new_xeno
+		if(length(xenomorphs) >= xeno_starting_num)
+			break
+
+	if(!length(xenomorphs))
+		return FALSE
+
+	xeno_required_num = CONFIG_GET(number/min_xenos)
+
+	if(length(xenomorphs) < xeno_required_num)
+		for(var/i = 1 to xeno_starting_num - length(xenomorphs))
+			new /mob/living/carbon/xenomorph/larva(pick(GLOB.xeno_spawn))
+
+	else if(length(xenomorphs) < xeno_starting_num)
+		var/datum/hive_status/normal/HN = GLOB.hive_datums[XENO_HIVE_NORMAL]
+		HN.stored_larva += xeno_starting_num - length(xenomorphs)
+
+	return TRUE
+
+
+// Overrides
+
+/datum/game_mode/crash/job_after_spawn(mob/living/carbon/human/H, mob/M, latejoin = FALSE)
```

Comment:
If they spawn in another squad, won't they get the squad messsage twice? This does not look like the best way to do this.

Reply:
Agreed, they do, i don't know a better way yet,

Reply:
Perhaps overriding it via ticker or where it's first called? You will have to research how job code works to do it properly, we don't want to introduce more snowflaking.

Reply:
I've added this as a task at the top once thats ticked we can revisit this but resolving for now.